File: openharmony-sig/ohos_react_native/tester/harmony/react_native_openharmony/src/main/ets/RNOHCorePackage/turboModules/VibrationTurboModule.ts
/**
 * Copyright (c) 2024 Huawei Technologies Co., Ltd.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE-MIT file in the root directory of this source tree.
 */

import type { TurboModuleContext } from '../../RNOH/TurboModule';
import { TurboModule } from '../../RNOH/TurboModule';
import { RNOHLogger } from '../../RNOH/RNOHLogger';
import { VibrationController } from '../../RNOH/VibrationController';

export class VibrationTurboModule extends TurboModule {
  public static readonly NAME = 'Vibration';

  private logger: RNOHLogger;
  private vibrationController: VibrationController

  constructor(protected ctx: TurboModuleContext) {
    super(ctx);
    this.logger = this.ctx.logger.clone("VibrationTurboModule")
    this.vibrationController = new VibrationController(this.logger)
  }

  public vibrate(pattern: number) {
    this.vibrationController.vibrate(pattern);
  };

  // Android only
  public vibrateByPattern(pattern: Array<number>, repeat: number) {
    this.logger.warn("Vibration::vibrateByPattern is not supported");
  };

  public cancel() {
    this.vibrationController.cancel();
  }
}

File: openharmony-sig/ohos_react_native/tester/harmony/react_native_openharmony/src/main/cpp/RNOHCorePackage/TurboModules/VibrationTurboModule.cpp
/**
 * Copyright (c) 2024 Huawei Technologies Co., Ltd.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE-MIT file in the root directory of this source tree.
 */

#include "RNOHCorePackage/TurboModules/VibrationTurboModule.h"

using namespace facebook;

namespace rnoh {
VibrationTurboModule::VibrationTurboModule(
    const ArkTSTurboModule::Context ctx,
    const std::string name)
    : ArkTSTurboModule(ctx, name) {
  methodMap_ = {
      ARK_METHOD_METADATA(vibrate, 1),
      ARK_METHOD_METADATA(cancel, 0),
  };
}
} // namespace rnoh
